src/release_notes_workflow/ai_release_notes_generator.py

```diff
@@ -0,0 +1,499 @@
+   - Always wrap PR links in parentheses
+   - **IMPORTANT**: Capitalize the first letter of every entry description (after the asterisk and space)
+
+4. **Output Requirements:**
+   - Generate markdown with ## headers for each category
+   - Only include categories that have entries
+   - Sort categories in this order: Breaking Changes, Features, Enhancements, Bug Fixes, Infrastructure, Documentation, Maintenance, Refactoring, Other
+   - Each entry should be a single line with proper PR link formatting
+
+5. **PR Link Format:**
+   - Extract PR number from commit message
+   - Format as: `([#<number>]({repository_url}/pull/<number>))`
+   - Example: `([#456]({repository_url}/pull/456))`
+   
+6. **Important Notes:**
+   - Every commit should be categorized into exactly one category
+   - If you cannot determine the appropriate category from labels OR content analysis, place the entry in an "Other" category
+   - Do not skip any commits - every entry must appear somewhere in the release notes
+   - Prioritize GitHub PR labels over commit message analysis when available
+
+Generate the release notes in category-first format like this example:
+## Features
+* Feature description ([#123]({repository_url}/pull/123))
+* Another feature description ([#124]({repository_url}/pull/124))
+
+## Enhancements
+* Enhancement description ([#125]({repository_url}/pull/125))
+* Another enhancement description ([#126]({repository_url}/pull/126))
+
+## Bug Fixes
+* Bug fix description ([#127]({repository_url}/pull/127))"""
+        
+        return prompt
+    
+    def _write_release_notes_to_repo(self, repo: GitRepository, component_name: str, content: str):
+        """Write release notes to repository's release-notes directory."""
+        release_notes_dir = os.path.join(repo.dir, "release-notes")
+        os.makedirs(release_notes_dir, exist_ok=True)
+        
+        filename = f"{component_name.lower()}.release-notes-{self.version}.md"
+        filepath = os.path.join(release_notes_dir, filename)
+        
+        with open(filepath, 'w') as f:
+            f.write(f"# {component_name} {self.version} Release Notes\n\n")
+            f.write(content)
+        
+        logging.info(f"Wrote release notes to {filename}")
+    
+    def _commit_release_notes(self, repo: GitRepository, component_name: str):
+        """Commit release notes changes."""
+        try:
+            # Create release branch
+            branch_name = f"{self.version}-release-notes"
+            repo.execute(f"git checkout -b {branch_name}")
+            
+            # Add and commit
+            repo.execute("git add release-notes/")
+            repo.execute(f'git commit -m "Add {self.version} release notes for {component_name}"')
+            
+            logging.info(f"Committed release notes on branch {branch_name}")
+            
+        except Exception as e:
+            logging.warning(f"Failed to commit release notes: {e}")
```

Comment:
@rishabh6788 Since we already have the token for retrieval wondering do we want to generate PR from here or let CI take care of it?

Reply:
Just realized that it is easier in Ci because everything runs in the context of build repo as final release notes PR is raised in the build repo, however, here we will be creating PRs for each component repo, which is getting checked out in the context of GitRepository python class and gets deleted as soon as processing is over, it is much easier to add pull request logic in python code than jenkins.